### ISSUE TITLE
Undefined behavior in va_start, compilation error

### ISSUE THREAD

**Issue by AMDmi3** [fuzzylite/fuzzylite#77]
clang 4.0.0:

```
/tmp/fuzzylite/fuzzylite/./fl/Operation.h:1044:24: error: passing an object that undergoes default argument promotion to 'va_start' has undefined behavior [-Werror,-Wvarargs]
        va_start(args, first);
                       ^
/tmp/fuzzylite/fuzzylite/./fl/Operation.h:1039:19: note: parameter of type 'float' is declared here
            float first, ...) {
                  ^
```

**Comment by jcrada** [comment]
Thanks for reporting this. I am still looking into a solution for this compilation warning, but my current research is pointing to just adding "-Wno-varargs" to the CMAKE_CXX_FLAGS in cmake to ignore the warning.

**Comment by AMDmi3** [comment]
Yes, that's what we have to do for FreeBSD port of VCMI which bundles fuzzylite. However, UB is still an UB, it may not work correctly as it is or it will break on the next clang update.

**Comment by jcrada** [comment]
Well, I understand the logic to mark this as a Undefined Behaviour (UB), even though it is not undefined behaviour. The argument in the method is a `float`, which is automatically promoted to a `double` because of the variadic definition, but since `float != double` then it is marked as UB. 

The problem is that I cannot find a way to have a variadic `float` method without raising a warning.

After spending hours trying to find a workaround, (1) I removed the compiler flag to treat warnings as errors (which I should have removed before release anyway), and (2) I added the `-Wno-varargs` to the CMake list, but I wish I did not have to do this. I will have this in mind for next version.
